### PR TITLE
refactor demangling part in profiling.rs

### PR DESCRIPTION
**Pull request by high-cloud**
…r_index

issue #7665
replace the match statement with a function already exist

### CODE REVIEW

File: crates/wasmtime/src/profiling.rs
Comment (fitzgen) [inline crates/wasmtime/src/profiling.rs:196]: Could we change the `unwrap` to an `expect` where there is a brief message explaining why we expect this never to fail? Something like

```suggestion
        .expect("only errors if the Writer errors, and string writers never error");
```

**Review by fitzgen — APPROVED** [review]
LGTM with minor nitpick below. Thanks!

**Comment by elliottt** [comment]
Would you mind rebasing this PR? Once that's done, I'm happy to click the merge button :+1:

**Pull request merged**
